### PR TITLE
Add changelog for 1.16.0

### PR DESCRIPTION
**Pull request by mAAdhaTTah**
This is WIP, as I'll be fixing it as new PRs come in. There are a few I want to land before we release, but this gets it started so we can finally get a release out.

Also modified the changelog generation, as I'd like to try and automate as much of this as possible, so we can get releases out more frequently than we have been.

### CODE REVIEW

**Comment by mAAdhaTTah** [comment]
2 notes:

1. Created a milestone for 1.16.0 for the last few things I'd like to land before we release.
2. I'm going to rebase these commits after I'm done, but I'll be pushing incrementally.

**Comment by RunDevelopment** [comment]
I wasn't aware of this before, but we should also adjust the `.npmignore` file because of #1781.

**Comment by mAAdhaTTah** [comment]
Rebased & ready for review.

**Comment by RunDevelopment** [comment]
Unrelated: I know what [the first thing I will do](https://gulpjs.com/docs/en/getting-started/javascript-and-gulpfiles#splitting-a-gulpfile) after we publish 1.16.0.

**Comment by mAAdhaTTah** [comment]
Good call. While sorting, I also realized I had Java twice. I also love that we have both `Markup` & `Markdown` as languages 😆.

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,264 @@
...
+
+* Normalized the font-size of pre and code ([#1791](https://github.com/PrismJS/prism/issues/1791)) [`878ef295`](https://github.com/PrismJS/prism/commit/878ef295)
+* __Coy__
+	* Correct typo ([#1508](https://github.com/PrismJS/prism/issues/1508)) [`c322fc80`](https://github.com/PrismJS/prism/commit/c322fc80)
+
+### Other changes
+
+* __Core__
+	* `insertBefore` now correctly updates references ([#1531](https://github.com/PrismJS/prism/issues/1531)) [`9dfec340`](https://github.com/PrismJS/prism/commit/9dfec340)
+	* Replaced all occurrences of `new RegExp` with `RegExp` ([#1493](https://github.com/PrismJS/prism/issues/1493)) [`44fed4d3`](https://github.com/PrismJS/prism/commit/44fed4d3)
```
Comment (RunDevelopment) [inline CHANGELOG.md]: A few of the items under __Core__ don't really belong in any category, so why not move them to the top of __Other changes__, similar to the uncategorized changes in __Updated components__?

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,264 @@
 # Prism Changelog
 
+## 1.16.0 (2019-03-24)
+
+### New components
+
+* __GameMaker Language__ ([#1551](https://github.com/PrismJS/prism/issues/1551)) [`e529edd8`](https://github.com/PrismJS/prism/commit/e529edd8)
+* __CIL__ ([#1593](https://github.com/PrismJS/prism/issues/1593)) [`38def334`](https://github.com/PrismJS/prism/commit/38def334)
```
Comment (RunDevelopment) [inline CHANGELOG.md]: I think these should be sorted as well.

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,264 @@
...
+	* Add command to generate CHANGELOG [`212666d3`](https://github.com/PrismJS/prism/commit/212666d3)
+	* Name in composer.json set to lowercase ([#1824](https://github.com/PrismJS/prism/issues/1824)) [`4f78f1d6`](https://github.com/PrismJS/prism/commit/4f78f1d6)
+	* Added alias tests ([#1832](https://github.com/PrismJS/prism/issues/1832)) [`5c1a6fb2`](https://github.com/PrismJS/prism/commit/5c1a6fb2)
+	* Travis: Fail when changed files are detected ([#1819](https://github.com/PrismJS/prism/issues/1819)) [`66b44e3b`](https://github.com/PrismJS/prism/commit/66b44e3b)
+	* Tests: Additional checks for Prism functions ([#1803](https://github.com/PrismJS/prism/issues/1803)) [`c3e74ea3`](https://github.com/PrismJS/prism/commit/c3e74ea3)
+	* Adjusted .npmignore ([#1834](https://github.com/PrismJS/prism/issues/1834)) [`29a30c62`](https://github.com/PrismJS/prism/commit/29a30c62)
+* __Website__
+	* Add Python triple-quoted strings "known failure" ([#1449](https://github.com/PrismJS/prism/issues/1449)) [`334c7bca`](https://github.com/PrismJS/prism/commit/334c7bca)
```
Comment (RunDevelopment) [inline CHANGELOG.md]: Should changes to examples really be under __Website__? [Previously](https://github.com/PrismJS/prism/blob/d455c437086444dd32a47310190179ee302bce0f/CHANGELOG.md#other-changes-2) these changes were uncategorized but maybe we could also add an __Examples__ category?

File: CHANGELOG.md (hunk 1 above)
Comment (mAAdhaTTah) [reply CHANGELOG.md]: Besides this, any other items in particular?

Comment (mAAdhaTTah) [reply CHANGELOG.md]: Actually, I think that and the `components.json` should be it.

File: CHANGELOG.md (hunk 3 above)
Comment (mAAdhaTTah) [reply CHANGELOG.md]: I don't think an extra category is necessary. The Examples page is part of the website, so it seems reasonable to include here.

File: CHANGELOG.md (hunk 2 above)
Comment (mAAdhaTTah) [reply CHANGELOG.md]: Done.

File: CHANGELOG.md (hunk 1 above)
Comment (RunDevelopment) [reply CHANGELOG.md]: Here are the PR ids: `(#1493) (#1503) (#1804)`

Comment (mAAdhaTTah) [reply CHANGELOG.md]: Added.

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,5 +1,264 @@
...
+
+### Updated components
+
+* Fixed dependencies of Pug and Pure ([#1759](https://github.com/PrismJS/prism/issues/1759)) [`c9a32674`](https://github.com/PrismJS/prism/commit/c9a32674)
+* Add file extensions support for major languages ([#1478](https://github.com/PrismJS/prism/issues/1478)) [`0c8f6504`](https://github.com/PrismJS/prism/commit/0c8f6504)
+* Fixed patterns which can match the empty string ([#1775](https://github.com/PrismJS/prism/issues/1775)) [`86dd3e42`](https://github.com/PrismJS/prism/commit/86dd3e42)
+* More variables for better code compression ([#1489](https://github.com/PrismJS/prism/issues/1489)) [`bc53e093`](https://github.com/PrismJS/prism/commit/bc53e093)
+* Added missing aliases ([#1830](https://github.com/PrismJS/prism/issues/1830)) [`8d28c74c`](https://github.com/PrismJS/prism/commit/8d28c74c)
+* Replaced all occurrences of `new RegExp` with `RegExp` ([#1493](https://github.com/PrismJS/prism/issues/1493)) [`44fed4d3`](https://github.com/PrismJS/prism/commit/44fed4d3) ([#1503](https://github.com/PrismJS/prism/issues/1503)) ([#1804](https://github.com/PrismJS/prism/issues/1804))
```
Comment (RunDevelopment) [inline CHANGELOG.md]: What happened here?

Comment (mAAdhaTTah) [reply CHANGELOG.md]: Misunderstood your comment. Will fix.

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,5 +1,264 @@
...
+	* Improved JavaScript parameter recognization ([#1722](https://github.com/PrismJS/prism/issues/1722)) [`57a92035`](https://github.com/PrismJS/prism/commit/57a92035)
+	* Make `undefined` a keyword in JS ([#1740](https://github.com/PrismJS/prism/issues/1740)) [`d9fa29a8`](https://github.com/PrismJS/prism/commit/d9fa29a8)
+	* Fix `function-variable` in JS ([#1739](https://github.com/PrismJS/prism/issues/1739)) [`bfbea4d6`](https://github.com/PrismJS/prism/commit/bfbea4d6)
+	* Improved JS constant pattern ([#1737](https://github.com/PrismJS/prism/issues/1737)) [`7bcec584`](https://github.com/PrismJS/prism/commit/7bcec584)
+	* Improved JS function pattern ([#1736](https://github.com/PrismJS/prism/issues/1736)) [`8378ac83`](https://github.com/PrismJS/prism/commit/8378ac83)
+	* JS: Fixed variables named "async" ([#1738](https://github.com/PrismJS/prism/issues/1738)) [`3560c643`](https://github.com/PrismJS/prism/commit/3560c643)
+	* JS: Keyword fix ([#1808](https://github.com/PrismJS/prism/issues/1808)) [`f2d8e1c7`](https://github.com/PrismJS/prism/commit/f2d8e1c7)
+* __JSON__ / __JSONP__
+	* Fix bugs in JSON language [`74fe81c6`](https://github.com/PrismJS/prism/commit/74fe81c6)
```
Comment (RunDevelopment) [inline CHANGELOG.md]: I know it's not part of the commit history, but we still might want to link to the PR `#1479`.

File: CHANGELOG.md (hunk 4 above)
Comment (mAAdhaTTah) [reply CHANGELOG.md]: I think `#1804` should stay in Core. It changed core as well as the components.

File: CHANGELOG.md (hunk 5 above)
Comment (mAAdhaTTah) [reply CHANGELOG.md]: Fixed.

**Pull request merged**